pdns/dnsdistdist/dnsdist.cc

```diff
@@ -1720,9 +1720,13 @@ bool assignOutgoingUDPQueryToBackend(std::shared_ptr<DownstreamState>& downstrea
   bool doh = dnsQuestion.ids.du != nullptr;
 
   bool failed = false;
+  dnsQuestion.ids.d_proxyProtocolPayloadSize = 0;
   if (downstream->d_config.useProxyProtocol) {
     try {
-      addProxyProtocol(dnsQuestion, &dnsQuestion.ids.d_proxyProtocolPayloadSize);
+      size_t proxyProtocolPayloadSize = 0;
+      if (addProxyProtocol(dnsQuestion, &proxyProtocolPayloadSize)) {
+        dnsQuestion.ids.d_proxyProtocolPayloadSize = proxyProtocolPayloadSize;
```

Comment:
I'd rather write
```suggestion
        dnsQuestion.ids.d_proxyProtocolPayloadSize += proxyProtocolPayloadSize;
```
for future-proofness.

Reply:
I'm afraid I disagree, `+=` sounds like we could be adding to an existing value, and this is not what is happening.

Reply:
You are adding to an existing value of zero...

Reply:
Technically, I'm replacing the existing value :) But I don't care that much, I'll make the change.

Reply:
I don't see it happen, since we always construct the payload in one go, but my crystal ball is not working as the company went bankrupt, so I applied your suggested change!

Reply:
That'll teach you to buy one of these new, Internet-connected, crystal balls. "Internet of Truth" they call'em. And then the product is discontinued or the company disappears, and you can't anything in it anymore.